packages/react-dom/src/__tests__/ReactUpdates-test.js

```diff
@@ -48,306 +48,395 @@ describe('ReactUpdates', () => {
     );
   }
 
-  it('should batch state when updating state twice', () => {
-    let updateCount = 0;
-
-    class Component extends React.Component {
-      state = {x: 0};
-
-      componentDidUpdate() {
-        updateCount++;
-      }
+  it('should batch state when updating state twice', async () => {
```

Comment:
While I was here, I rewote any of the class component tests that don't rely on class component features (like setState callbacks) to function components.